boards/arm/bt510/doc/bt510.rst

```diff
@@ -0,0 +1,219 @@
+*************************
+
+Applications for the ``bt510`` board configuration can be
+built and flashed in the usual way (see :ref:`build_an_application`
+and :ref:`application_run` for more details); however, the standard
+debugging targets are not currently available.
+
+The BT510 features a TagConnect 10 way socket for connection of a
+programmer/debugger, refer to `TagConnect TC2050 product page`_
+for details of an appropriate TagConnect cable.
+
+A non-standard layout is used to include access to the sensor debug UART.
+
++-----------+------------+----------------------+
+| Pin No.   | Name       | Description          |
++===========+============+======================+
+| 1         | Vcc        | Power Supply, 3.3V   |
++-----------+------------+----------------------+
+| 2         | SWDIO      | SWD Data             |
++-----------+------------+----------------------+
+| 3         | RXD        | Debug UART RX Data   |
++-----------+------------+----------------------+
+| 4         | SWDCLK     | SWD Clock            |
++-----------+------------+----------------------+
+| 5         | TM         | Spare GPIO           |
++-----------+------------+----------------------+
+| 6         | SWO        | SWD Output           |
++-----------+------------+----------------------+
+| 7         | N/C        | Not Connected        |
++-----------+------------+----------------------+
+| 8         | TXD        | Debug UART TX Data   |
++-----------+------------+----------------------+
+| 9         | GND        | Ground               |
++-----------+------------+----------------------+
+| 10        | RESET      | Reset, Active Low    |
++-----------+------------+----------------------+
+
+Connectivity to the programmer/debugger must be modified to match
+the pinout shown above.
+
+Laird Connectivity provide the USB-SWD programming board that supports
+this connector layout, refer to the `USB SWD Programmer product page`_
+.
+
+Flashing
+========
+
+Follow the instructions in the :ref:`nordic_segger` page to install
+and configure all the necessary software. Further information can be
+found in :ref:`nordic_segger_flashing`. Then build and flash
+applications as usual (see :ref:`build_an_application` and
+:ref:`application_run` for more details).
+
+Here is an example for the :ref:`hello_world` application.
+
+First, run your favorite terminal program to listen for output.
+
+NOTE: On the BT510, the FTDI USB should be used to access the UART console.
+
+.. code-block:: console
+
+   $ minicom -D <tty_device> -b 115200
+
+Replace :code:`<tty_device>` with the port where the bt510 can be found. For example, under Linux, :code:`/dev/ttyUSB0`.
+
+Then build and flash the application in the usual way.
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/hello_world
+   :board: bt510
+   :goals: build flash
+
+Debugging
+=========
+
+Refer to the :ref:`nordic_segger` page to learn about debugging Nordic based boards with a
+Segger IC.
```

Comment:
It requires an external programmer, it does not have an on-board one

Reply:
Updated.